chainer/links/activation/dropconnect.py

```diff
@@ -0,0 +1,94 @@
+import math
+
+from chainer import cuda
+from chainer.functions.noise import dropconnect
+from chainer import initializers
+from chainer import link
+
+
+class Dropconnect(link.Link):
+
+    """Fully-connected layer with dropconnect regularization.
+
+    Args:
+        in_size (int): Dimension of input vectors. If None, parameter
+            initialization will be deferred until the first forward data pass
+            at which time the size will be determined.
+        out_size (int): Dimension of output vectors.
+        wscale (float): Scaling factor of the weight matrix.
+        initialW (3-D array or None): Initial weight value.
+            If ``None``, then this function uses ``wscale`` to initialize.
+        initial_bias (2-D array, float or None): Initial bias value.
+            If it is float, initial bias is filled with this value.
+            If it is ``None``, bias is omitted.
+        reuse_mask (boolean):
+            If ``True``, once dropconnect mask is generated,
+            same mask will be reuse when link is called.
```

Comment:
As you said, `dropconnect` mask must not be fixed in normal use.
`reuse_mask` is only for paasing `check_backward`, but dirty, not preferable API.

Your idea's exactly what I wanted!